File: 2025-11/2025-11-11/vowels_and_consonants.py
import re

def count(s):

    return [
        len(re.findall("[aeiou]", s, re.IGNORECASE)),
        len(re.findall("[b-df-hj-np-tv-z]", s, re.IGNORECASE))
    ]

print(count("Hello World"))
print(count("JavaScript"))
print(count("Python"))
print(count("freeCodeCamp"))
print(count("Hello, World!"))
print(count("The quick brown fox jumps over the lazy dog."))

print(count("xyz"))
print(count("oua"))
print(count("!@#"))

# --- TEST SUITE ---

tests_text = '''
1. count("Hello World") should return [3, 7].
2. count("JavaScript") should return [3, 7].
3. count("Python") should return [1, 5].
4. count("freeCodeCamp") should return [5, 7].
5. count("Hello, World!") should return [3, 7].
6. count("The quick brown fox jumps over the lazy dog.") should return [11, 24].
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+?)\.?$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("--------------------------")
    print("🧪 Starting Verification...")
    print("--------------------------")

    all_passed = True
    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            all_passed = False
            fail_count += 1

    print("----------------------------");

    if (all_passed):
        print("🎉 SUCCESS: All tests PASSED.");
    else:
        print(f"⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED.");

run_tests(test_data)

File: 11 November/11-11/vowelsAndConsonants.js
function count(str) {

  return [
    str.match(/[aeiou]/gi)?.length ?? 0, 
    str.match(/[b-df-hj-np-tv-z]/gi)?.length ?? 0
  ];
}

console.log(count("Hello World"));
console.log(count("JavaScript"));
console.log(count("Python"));
console.log(count("freeCodeCamp"));
console.log(count("Hello, World!"));
console.log(count("The quick brown fox jumps over the lazy dog."));

console.log(count("xyz"));
console.log(count("oua"));
console.log(count("!@#"));

// --- TEST SUITE ---

const testsText = `
1. count("Hello World") should return [3, 7].
2. count("JavaScript") should return [3, 7].
3. count("Python") should return [1, 5].
4. count("freeCodeCamp") should return [5, 7].
5. count("Hello, World!") should return [3, 7].
6. count("The quick brown fox jumps over the lazy dog.") should return [11, 24].
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+?)\.?$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("--------------------------");
  console.log("🧪Starting Verification...");
  console.log("--------------------------");

  let allPassed = true;
  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(test.output);
    let comparison;

    if (Array.isArray(testOutput))
      comparison = arraysEqual(functionCallOutput, testOutput);
    else
      comparison = functionCallOutput === testOutput;

    if (comparison) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    }
    else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${testOutput}\nGot: ${functionCallOutput}`);
      allPassed = false;
      failCount ++;
    }
  })

  console.log("----------------------------");

  if (allPassed)
    console.log("🎉SUCCESS: All tests PASSED.");
  else
    console.log(`⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`);
}

function arraysEqual(a, b) {
  if (a.length !== b.length)
    return false;

  for (const index in a) {
    if (Array.isArray(a[index])) {
      if (Array.isArray(b[index])) {
        if (!arraysEqual(a[index], b[index]))
          return false;
      }
      else 
        return false;
    }

    else if (a[index] !== b[index])
      return false;
  }

  return true;
}

runTests(testData);